pkg/controller/datavolume/util.go

```diff
@@ -129,21 +128,22 @@ func pvcFromStorage(client client.Client, recorder record.EventRecorder, log log
 func renderPvcSpecVolumeModeAndAccessModesAndStorageClass(client client.Client, recorder record.EventRecorder, log *logr.Logger,
 	dv *cdiv1.DataVolume, pvcSpec *v1.PersistentVolumeClaimSpec, dvContentType cdiv1.DataVolumeContentType) error {
 	logInfo := func(msg string, keysAndValues ...interface{}) {
-		if log != nil {
-			log.V(1).Info(msg, keysAndValues...)
+		if log != nil && dv != nil {
```

Comment:
not sure I get this changes.. In some places you are not logging anything if dv doesnt exists and in others you are removing any information related to dv which can be useful? seems odd

Reply:
In the PVC webhook (`cdi-apiserver`) we don't have `logr.Logger` (unlike in `cdi-deployment`), but `klog` which has different api, and I see no reason to log there as we return the failure anyway to the user when she creates the PVC. Clearly, we don't have a DV there (it's a PVC webhook...), and no event recorder. However I think we should keep using `renderPvcSpecVolumeModeAndAccessModesAndStorageClass()` (don't complain about the name, it's Michael's afair) func from both DV controller and PVC webhook rendering. wdyt?

Reply:
btw, it's still wip as I need to better explain the issue and add utests.

Reply:
Oh I missed the fact that all this is inside functions so the dv data is kept. So ok looks good :) If you would have explained it in the commit message it would have been clearer